src/NULS/Signer.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+
+#include "Address.h"
+#include "BinaryCoding.h"
+#include "../Hash.h"
+#include "../PrivateKey.h"
+
+using namespace TW;
+using namespace TW::NULS;
+
+Signer::Signer(Proto::SigningInput& input) : input(input) {
+    Proto::TransactionCoinFrom coinFrom;
+    coinFrom.set_from_address(input.from());
+    coinFrom.set_assets_chainid(input.chain_id());
+    coinFrom.set_assets_id(input.idassets_id());
+    //need to update with amount + fee
+    coinFrom.set_id_amount(input.amount());
+    coinFrom.set_nonce(input.nonce());
+    //default unlocked
+    coinFrom.set_locked(0);
+    *tx.mutable_input() = coinFrom;
+
+    Proto::TransactionCoinTo coinTo;
+    coinTo.set_to_address(input.to());
+    coinTo.set_id_amount(input.amount());
+    coinTo.set_assets_chainid(input.chain_id());
+    coinTo.set_assets_id(input.idassets_id());
+    coinTo.set_lock_time(0);
+    *tx.mutable_output() = coinTo;
+
+    tx.set_remark(input.remark());
+    tx.set_type(2);
```

Comment:
what's this magic value?

Reply:
this is nuls 2.0 transaction type. I will define macro.